packages/cli/src/sso/saml/routes/saml.controller.ee.ts

```diff
@@ -70,36 +73,39 @@ export class SamlController {
 	 * GET /sso/saml/acs
 	 * Assertion Consumer Service endpoint
 	 */
-	@Get(SamlUrls.acs, { middlewares: [samlLicensedAndEnabledMiddleware] })
+	@Get(SamlUrls.acs, { middlewares: [samlLicensedMiddleware] })
 	async acsGet(req: express.Request, res: express.Response) {
-		const loginResult = await this.samlService.handleSamlLogin(req, 'redirect');
-		if (loginResult) {
-			if (loginResult.authenticatedUser) {
-				await issueCookie(res, loginResult.authenticatedUser);
-				if (loginResult.onboardingRequired) {
-					return res.redirect(SamlUrls.samlOnboarding);
-				} else {
-					return res.redirect(SamlUrls.defaultRedirect);
-				}
-			}
-		}
-		throw new AuthError('SAML Authentication failed');
+		return this.acsHandler(req, res, 'redirect');
 	}
 
 	/**
 	 * POST /sso/saml/acs
 	 * Assertion Consumer Service endpoint
 	 */
-	@Post(SamlUrls.acs, { middlewares: [samlLicensedAndEnabledMiddleware] })
+	@Post(SamlUrls.acs, { middlewares: [samlLicensedMiddleware] })
 	async acsPost(req: express.Request, res: express.Response) {
-		const loginResult = await this.samlService.handleSamlLogin(req, 'post');
+		return this.acsHandler(req, res, 'post');
+	}
+
+	/**
+	 * Handles the ACS endpoint for both GET and POST requests
+	 * Available if SAML is licensed, even if not enabled to run connection tests
+	 * For test connections, returns status 202 if SAML is not enabled
+	 */
+	async acsHandler(req: express.Request, res: express.Response, binding: SamlLoginBinding) {
+		const loginResult = await this.samlService.handleSamlLogin(req, binding);
 		if (loginResult) {
 			if (loginResult.authenticatedUser) {
-				await issueCookie(res, loginResult.authenticatedUser);
-				if (loginResult.onboardingRequired) {
-					return res.redirect(SamlUrls.samlOnboarding);
+				// Only sign in user if SAML is enabled, otherwise treat as test connection
+				if (isSamlLicensedAndEnabled()) {
```

Comment:
does it make sense to move this check above, so that if the check is `false`, we don't need to call `handleSamlLogin`.

Reply:
We do need to call handleSamlLogin, because in the case where the owner tests the authentication flow pre-enabling it, we need to be able to go through the full authentication steps and then return an OK, without actually signing them in.